flink-runtime/src/main/java/org/apache/flink/runtime/jobmanager/scheduler/Scheduler.java

```diff
@@ -152,12 +150,14 @@ else if (ret instanceof Future) {
 			}
 		}
 		catch (NoResourceAvailableException e) {
-			return FlinkCompletableFuture.completedExceptionally(e);
+			CompletableFuture<SimpleSlot> notEnoughResources = new CompletableFuture<>();
```

Comment:
I actually did in one of the other up PRs. I'll try to merge this one first and then adapt the code here.